presto-hive/src/main/java/com/facebook/presto/hive/HiveMetadata.java

```diff
@@ -1220,6 +1220,13 @@ public HiveInsertTableHandle beginInsert(ConnectorSession session, ConnectorTabl
         Map<List<String>, ComputedStatistics> partitionComputedStatistics = createComputedStatisticsToPartitionMap(computedStatistics, partitionedBy, columnTypes);
 
         for (PartitionUpdate partitionUpdate : partitionUpdates) {
+            if (partitionUpdate.getFileNames().size() == 0) {
+                HiveWriteUtils.createDirectory(
```

Comment:
Static import

Reply:
I don't think creating the directory here is there correct place. We should be doing it in `SemiTransactionalHiveMetastore` as part of the commit. We probably also need to handle it in the rollback cleaup. @haozhun should take a look